### DISCUSSION TITLE
Does Identity Core support for multi type user implementation ?

### DISCUSSION THREAD

**Discussion by slnkykrn** [dotnet/aspnetcore#28010] [Q&A] [answered]
Hello ,

We want to implement multi type user implementation with identity core for multi tenancy. In google solutions generally goes over one type of user or one ApplicationUser. We tried to build something like this but we had issues on expecially startup :

  public class ApplicationUser : IdentityUser
    {
    }

   public class Tenant1User: ApplicationUser 
    {
        public int deneme { get; set; }
    }

 public class Tenant2User: ApplicationUser 
    {
        public string City { get; set; }
        public string Picture { get; set; }
        public DateTime? BirthDay { get; set; }
        public int Gender { get; set; }
    }

For these type of implementation Identity allow just one type UserManager  or UserStore in startup but we need something like below in startup but these type of configuration is not allowed by identity: 

        .AddUserManager<UserManager<Tenant1User>>()
        .AddUserManager<UserManager<Tenant2Userr>>()

How could be  more than one customized user for UserManager or UserStore thing with identity core ? Expecially for startup configuration.

Thank you.

**Comment by blowdart** [comment]
No it does not. It does not support multi-tenancy at all.

**Comment by davidfowl** [comment, marked as answer]
**EDIT:**

Actually taking a deeper look it seems like we have some limitations that prevent this from working smoothly. We currently register these types concretely when you call AddIdentity. It doesn't look like there's anything fundamental that is preventing it from working though.

cc @HaoK

**Reply by davidfowl** [reply]
You might need to call AddIdentity or AddIdentityCore for your second tenant type:

```C#
AddIdentityCore<Tenant2Userr>()
```
